app/api/routes/product.py

```diff
@@ -0,0 +1,138 @@
+from fastapi import APIRouter, Depends, HTTPException, status, Response, Query
+from sqlalchemy.orm import Session
+from app.db.database import get_db
+from app.schemas.product import ProductCreate, ProductResponse, ProductUpdate
+from app.api.services.product_service import ProductService
+from app.api.exceptions.GlobalException import (
+    ProductNotFoundException,
+    ProductAlreadyExistsException,
+    InvalidProductDataException,
+)
+from app.api.dependencies.ProductValidator import ProductValidator
+from uuid import UUID
+from app.models import Product, ProductSearchParams
+from typing import Optional, List
+from decimal import Decimal
+from sqlalchemy import asc, desc
+
+router = APIRouter()
+
+
+@router.post(
+    "/products/", response_model=ProductResponse, status_code=status.HTTP_201_CREATED
+)
+def create_product(product_data: ProductCreate, db: Session = Depends(get_db)):
+    validator = ProductValidator(db)
+
+    try:
+        validator.validate_unique_name(product_data.name)
+        validator.validate_product_data(product_data)
+
+        new_product = Product(**product_data.dict())
```

Comment:
This and the next three lines, it's cleaner to move them to a new method in the productservice like create_product

Reply:
I think you already have it